apis/mattermost/v1beta1/mattermost_types.go

```diff
@@ -364,6 +367,15 @@ type ExternalFileStore struct {
 	Secret string `json:"secret,omitempty"`
 }
 
+// ExternalVolumeFileStore defines the configuration of an externally managed
+// volume file store.
+type ExternalVolumeFileStore struct {
+	// The name of the externally managed volume to be mounted for file storage.
+	VolumeName string `json:"volumeName,omitempty"`
```

Comment:
Do we require the `VolumeName` parameter? We can rely on `VolumeClaimName`, and if the status is `Bound`, it is safe to mount the volume. We can set `external-volume` as `Volume` name in Deployment.

Reply:
Hmm interesting. You are right that we probably don't explicitly need a separate name value. Let me think about whether it's a good idea to keep it or not. @fmartingr have any input here?

Reply:
PV is usually managed by storageclass. As a result, the name is entirely arbitrary. Thats why, I believe it will be difficult to use the same manifest several times in several places.

Reply:
> Thats why, I believe it will be difficult to use the same manifest several times in several places.

Sorry could you help me understand what you mean by this? Are you saying that by allowing for arbitrary names there is more chance to pass an incorrect value into a resource manifest?

Reply:
When we create a PVC, a PV with a random name is created automatically. Lets say we have a `Mattermost` manifest like below

```
fileStore:
  externalVolume:
    volumeClaimName: mm-data
    volumeName: pvc-<random string>
```

If we want to provide `volumeName`, we need to change this field every time for a new installation.  That means, If we use same manifest multiple times without setting `volumeName` to the correct one, yes, It will hold incorrect value.

Reply:
Okay I think I understand now. The intent of this filestore option is to provide the most flexibility possible for admins who need to deploy Mattermost with our non-S3 options.

> If we want to provide volumeName, we need to change this field every time for a new installation

Yeah, that's correct. I am less concerned about needing to provide specific configuration for each installation as much of our configuration is like that and the Cloud Provisioner handles that for us for most stuff. That said, we won't be actively using this filestore option ourselves and is primarily made for admins using the operator in other environments. In those cases they would only need to setup and configure everything once so it shouldn't be an issue.

All that said, I am generally in favor of minimal configuration so let's remove the volume name for now. We can always add it back later if we need to. 👍